src/errors.rs

```diff
@@ -1,61 +1,110 @@
 // Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
-use thiserror::Error;
```

Comment:
After this change is `thiserror` used at all? Could you get rid of that dependency altogether?

(Or, if it is used somewhere else, why is that "somewhere else" not breaking `no_std`?)